components/base/sensorcontrolled/movestraight.go

```diff
@@ -0,0 +1,191 @@
+// Package sensorcontrolled base implements a base with feedback control from a movement sensor
+package sensorcontrolled
+
+import (
+	"context"
+	"math"
+	"time"
+
+	geo "github.com/kellydunn/golang-geo"
+
+	"go.viam.com/rdk/components/movementsensor"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+const (
+	slowDownDistGain      = .1
+	maxSlowDownDist       = 100 // mm
+	moveStraightErrTarget = 20  // mm
+	headingGain           = 1.
+)
+
+func (sb *sensorBase) MoveStraight(
+	ctx context.Context, distanceMm int, mmPerSec float64, extra map[string]interface{},
+) error {
+	ctx, done := sb.opMgr.New(ctx)
+	defer done()
+	sb.setPolling(false)
```

Comment:
is setPolling necessary again? or is there still a plan to remove it in a future PR

Reply:
plan to remove in a future PR, since I left the logic in Spin